### PR TITLE
[WIP] Torch implementations (exact OT + Sinkhorn + 1D + sliced)

### PR DESCRIPTION
**Pull request by rflamary**
## Types of changes

- [x] New feature (non-breaking change which adds functionality)

What is planned in this PR:
- [ ] LP and sinkhorn solvers (takes and return  optionally detached torch tensors)
- [ ] LP and Sinkhorn loss/divergence functions that are differentiable wrt the weights and distance matrix
- [ ] `ot.torch.utils` torch utils functions for quick prototyping (dist, unif, proj_simplex)
- [ ] Proper documentation and examples

## Motivation and context / Related issue

Add a torch compatible functions for computing OT and OT loss for exact and sinkhorn divergences.

## Checklist

- [ ] The documentation is up-to-date with the changes I made.
- [x] I have read the [**CONTRIBUTING**](CONTRIBUTING.md) document.
- [ ] All tests passed, and additional code has been covered with new tests.

### CODE REVIEW

**Comment by AdrienCorenflos** [comment]
Hi,
I also have this code that I can contribute (opened a draft PR for it)
https://github.com/PythonOT/POT/pull/218
Let me know if you want me to iron it out or if you have other plans.
Adrien

**Comment by AdrienCorenflos** [comment]
Hi,
I just checked because of some jitting issues I was having, but pytorch doesn't officially support Python 3.5, so might be case for you to drop the support in POT too?

**Comment by rflamary** [comment]
This PR has been replaced by #249
